### ISSUE TITLE
Workqueues V2

### ISSUE THREAD

**Issue by rikukissa** [opencrvs/opencrvs-core#7073]
### Moved under #8445

**Comment by naftis** [comment]
@jpye-finch 

Currently we fetch all workqueues at the same time when the app is loaded.

Is there value in that or could we load each of them dynamically when that workqueue page is opened? This _could_ simplify the workqueue loading, just an thought at this point though. If you don't have internet you can see the workqueues you have opened earlier.

**Comment by jpye-finch** [comment]
Yes it could work. Would the experience not be same as we have to day? When you navigate between workqueues after login and initial load. You see the latest state of the workqueue whilst we check for changes. 

As long as you can still see your assigned records so you can work offline. We are good imo

**Comment by euanmillar** [comment]
This is also functional and based on experiment ascertain estimate.

**Comment by naftis** [comment]
@jpye-finch flagging this for ideation and design:

Frontend code logic - partly because of the current UX - around the downloaded records and fetched workqueues is confusing at points, as we want to keep the workqueues updated after a record leaves the outbox. There is no clear way to see which workqueue received the record other than understanding the backend business flow.

Could there be a concept of having a separate workqueue similar to Outbox for "My records". By default the records you've created are also downloaded and saved there? Then the workqueues themselves could be fairly online-dependent, while you could still work on your offline-workqueues Outbox and My records.

**Comment by jpye-finch** [comment]
@naftis thanks for this. 

I did have an 'Assigned to you' workqueue in the current epic description. 
https://www.figma.com/board/kbRxrSeitNzGahyojbdLM5/OpenCRVS-Roadmap?node-id=235-3617&t=Q3hnASzRS1WJvdAT-4

**Comment by jpye-finch** [comment]
@naftis 
Outbox and Assigned to you:

Conditional on the user having at least on of these scopes:

record.declare-{event}
record.declaration-send-for-approval
record.declaration-submit-for-updates
record.review-duplicates
record.register
record.declaration-archive
record.registration-reinstate
record.registration-revoke
record.print&issue-certified-copies
record.registration-correct
record.registration-request-correction

**Comment by jpye-finch** [comment]
@naftis I don't see in the dev tasks here to allow the icon of the workqueue to be configurable. Allowing you to set the icon from the Phosphor Icons library

**Comment by jpye-finch** [comment]
@naftis 
Also in this version will workqueues be based on scopes? Then we refactor so you can specify the custom users who have the workqueue?

users: ["Healthcare Worker", "Police Officer", "Registrar"]  
**instead of** 
requiredScopes": ["view-in-progress-records"],

**Comment by rikukissa** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/1322608?s=64&amp;v=4">@naftis](https://github.com/naftis) Also in this version will workqueues be based on scopes? Then we refactor so you can specify the custom users who have the workqueue?
> 
> users: ["Healthcare Worker", "Police Officer", "Registrar"] **instead of** requiredScopes": ["view-in-progress-records"],

@jpye-finch good question.. My knee jerk reaction is that we should define as much as we can with just raw scopes. "User types" would remain as a very thin light layer for labeling different user scope sets but wouldn't otherwise be referred to anywhere in configuration. 
This would make the workqueue drive by "you have access to feature X, thus you see this workqueue" and less "you are a Police officer, so you see this workqueue". 

I do see an issue here though as the workqueues are effectively just different types of advanced searches and not fully connected always to a functionality / scope. Makes me think a country should be able to specify a scope for a workqueue like `custom:view-childs-with-letter-a` which they could then also add to a user type they want.

**Comment by rikukissa** [comment]
@naftis this starts to look very good! One topic I'd still want you to consider – what's the upgrade story for countries upgrading from 1.6? In the custom scopes PR you might remember I had [this script](https://github.com/opencrvs/opencrvs-farajaland/pull/1042/files#diff-112b280e0b6d273ca833399a6bbbccf0d42f717e8d1d2a903f1fdb91d130ac93). 

Is it necessary as we know every country just has the basic set? Maybe we just add the code to CC and countries get it when they pull?

Feel free to mark as Ready to build after @jpye-finch signs it off. @Zangetsu101 I think someone can start working on this even before the scopes PR is merged, right? Just using the scopes we have today. Might require some ingenuity but still :)

**Comment by naftis** [comment]
@rikukissa I'd say we add the code to CC and the countries get it when they pull. I know there are custom workqueues implemented in some countries: we need documentation over how the workqueues can be configured after the technical work is done.

**Comment by rikukissa** [comment]
@naftis btw, I remember you started doing some exploration for this task – if you have any branches etc open, good to link them to the task description

**Comment by naftis** [comment]
Not really - the design became different as we decided to load all of the work queues' first pages simultaneously.

**Comment by anninawersun** [comment]
Uganda requirement: #6270 Senior Registration Officer user required

A Registrar plus:

Can review correction requests (if senior approval required)
Can review revoke and reinstatement requests (after review from the ED and Legal)

**Comment by rikukissa** [comment]
Blocking this ticket until it's either closed or adjusted to not include requirements already captured by https://github.com/opencrvs/opencrvs-core/issues/8445

**Comment by rikukissa** [comment]
All requirements except #8185 are captured by #8445. @jpye-finch this epic can be closed after the requirement is moved elsewhere.

**Comment by Erusha-Raju** [comment]
@jpye-finch thank you, will close 8445

**Comment by jpye-finch** [comment]
apologies @Erusha-Raju. I was mistaken! this ticket can be closed. Not https://github.com/opencrvs/opencrvs-core/issues/8445

**Comment by Erusha-Raju** [comment]
No problem, closing this issue

> apologies [@Erusha-Raju](https://github.com/Erusha-Raju). I was mistaken! this ticket can be closed. Not [#8445](https://github.com/opencrvs/opencrvs-core/issues/8445)
